src/Console/ConsoleOptionParser.php

```diff
@@ -387,7 +387,14 @@ public function addOption(ConsoleInputOption|string $name, array $options = [])
                 'required' => false,
                 'prompt' => null,
             ];
+
             $options += $defaults;
+
+            // If a default value is provided and it is not a string, do a cast
+            if ($options['default'] && !is_string($options['default'])) {
```

Comment:
```suggestion
            if ($options['default'] && is_int($options['default'])) {
```
sounds safer to me

Reply:
What about floats? They could be cast to string as well.

Reply:
Floats are usually already displayed within cake as strings, at least given the DB type etc.
So I think too much magic here isnt needed. The only common one people might trip over is int IMO.